test/widgets/inbox_test.dart

```diff
@@ -385,6 +450,50 @@ void main() {
           checkAppearsUncollapsed(tester, 1, findSectionContent);
         });
 
+        testWidgets('collapse stream section after scroll', (tester) async {
+          await setupVarious(tester);
+
+          final topicFinder = find.text('stream 1 topic 4' ).hitTestable();
+          final listFinder = find.byType(Scrollable);
+
+          // Scroll part of [_StreamSection] offscreen.
+          await dragUntilInvisible(
+            tester, topicFinder, listFinder, const Offset(0, -50));
+
+          final headerRow = findStreamHeaderRow(tester, 1);
+          // Check that the header is present (which must therefore
+          // be as a sticky header).
+          check(headerRow).isNotNull();
+
+          await tapCollapseIcon(tester, 1);
+
+          final headerRowAfterTap = findStreamHeaderRow(tester, 1);
+          // Check that the header is still visible even after
+          // collapsing the section.
+          check(headerRowAfterTap).isNotNull();
+        });
+
+        testWidgets('collapse stream section in the middle of screen', (tester) async {
```

Comment:
I think mentioning here the intended outcome (though very concisely) would help the reader understand the test. I suspect it'd have helped me when I overlooked this test previously: https://github.com/zulip/zulip-flutter/pull/560#discussion_r1536588587

So:
```suggestion
        testWidgets('collapse stream section in middle of screen: header stays fixed', (tester) async {
```